apps/pie-docs/src/_11ty/filters/headingAnchor.js

```diff
@@ -0,0 +1,35 @@
+const slugify = require('slugify');
```

Comment:
I'm interested to know why you chose to use a filter rather than an additional shortcode?

Can you remind me how we can use this granularly on a content page for some headings and not others please?

Reply:
I'm assuming you mean the anchor, not slugify?

The content page uses a markdown plugin so it only adds the anchor to markdown files.

We need a filter because we need to apply the anchor to the shortcodes. These don't have anchors automatically applied cos they're js, not md. We also wouldn't want them applied to every single shortcode because there are headings in shortcodes like `card` that we do not want the anchor on.

The filter can be applied directly to the shortcodes we want anchors on, `tokenTable` and `categorisedIconList`  as you can see here https://github.com/justeattakeaway/pie/pull/554/files#diff-6edac4cb8e6a5d69221c82c758f8476016f8228c2bd492055f947c6732efcfd2R56

I hope that answers your question

Reply:
Yeah it was regarding this entire file.

I was wondering if in an `md` file we'd ever want something like

```
{% anchorHeading { 'My Heading', '#some-link' %}
```

Added to the page directly?

Reply:
Maybe it would be useful at some point it might make sense to, but I think it makes more sense to automatically apply the anchor to all headings instead of manually adding this in instead of the headings

Reply:
Ok, so just to check I've understood - with this change are we only able to add anchor headings to shortcodes and not directly to markdown pages?

Reply:
@jamieomaguire The markdown filter has also been extended so it adds anchors there too

Reply:
Nope we have that automatically.

The markdown files will all automatically have the anchor added because of the markdown utility file
[this guy](https://github.com/justeattakeaway/pie/pull/554/files#diff-b14132ae0275f8dad8ab29be8a443846b462d85ed467be4bacf6d186e4ce0247R20)

The file this comment is on is the filter to filter the shortcodes that we want to have an anchor on, and add it to them 👍